tracer/src/Datadog.Trace/LibDatadog/CharSlice.cs

```diff
@@ -42,11 +43,16 @@ internal CharSlice(string? str)
         }
         else
         {
-            // copy over str to unmanaged memory
-            var bytes = System.Text.Encoding.UTF8.GetBytes(str);
-            Ptr = Marshal.AllocHGlobal(bytes.Length);
-            Marshal.Copy(bytes, 0, Ptr, bytes.Length);
-            Len = (nuint)bytes.Length;
+            var encoding = Encoding.UTF8;
```

Comment:
I wonder, should we be using `StringEncoding.UTF8` here so that we don't emit the BOM 🤔